src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MongoDbSessionHandler.php

```diff
@@ -132,19 +136,33 @@ public function gc($maxlifetime)
      */
     public function write($sessionId, $data)
     {
-        $expiry = new \MongoDate(time() + (int) ini_get('session.gc_maxlifetime'));
-
-        $fields = array(
-            $this->options['data_field'] => new \MongoBinData($data, \MongoBinData::BYTE_ARRAY),
-            $this->options['time_field'] => new \MongoDate(),
-            $this->options['expiry_field'] => $expiry,
-        );
-
-        $this->getCollection()->update(
-            array($this->options['id_field'] => $sessionId),
-            array('$set' => $fields),
-            array('upsert' => true, 'multiple' => false)
-        );
+        $expiry = $this->createDateTime(time() + (int) ini_get('session.gc_maxlifetime'));
+
```

Comment:
Yes.  I'd suggest having 2 different blocks due to the update method name changing too (update vs updateOne as well as needing the 'multiple' property passed in the options parameter for the first one.  I'll modify this to condense the $fields declaration though.